src/Microsoft.DotNet.Build.Tasks.Packaging/src/HarvestPackage.cs

```diff
@@ -0,0 +1,359 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks.Packaging
+{
+    public class HarvestPackage : PackagingTask
+    {
+        /// <summary>
+        /// Package ID to harvest
+        /// </summary>
+        [Required]
+        public string PackageId { get; set; }
+
+        /// <summary>
+        /// Current package version.
+        /// </summary>
+        [Required]
+        public string PackageVersion { get; set; }
+
+        /// <summary>
+        /// Folder where packages have been restored
+        /// </summary>
+        [Required]
+        public string PackagesFolder { get; set; }
+
+        /// <summary>
+        /// Path to runtime.json that contains the runtime graph.
+        /// </summary>
+        [Required]
+        public string RuntimeFile { get; set; }
+
+        /// <summary>
+        /// Additional packages to consider for evaluating support but not harvesting assets.
+        ///   Identity: Package ID
+        ///   Version: Package version.
+        /// </summary>
+        public ITaskItem[] RuntimePackages { get; set; }
+
+        /// <summary>
+        /// Set to false to suppress harvesting of files and only harvest supported framework information.
+        /// </summary>
+        public bool HarvestAssets { get; set; }
+
+        /// <summary>
+        /// Set to partial paths to exclude from file harvesting.
+        /// </summary>
+        public string[] PathsToExclude { get; set; }
+
+        /// <summary>
+        /// Set to partial paths to suppress from both file and support harvesting.
+        /// </summary>
+        public string[] PathsToSuppress { get; set; }
```

Comment:
Do you think it would make more sense if I had a parameter that controlled suppression of the harvested support frameworks rather than making someone realize that in order to suppress support for those they have to suppress the files that resulted in supporting that framework?